subsys/bluetooth/controller/ll_sw/ull_connections.c

```diff
@@ -223,7 +225,247 @@ void ll_rx_enqueue(struct node_rx_pdu *rx)
 	sys_slist_append(&ut_rx_q, (sys_snode_t *) rx);
 }
 
+/* TODO(tosk): Should be general utility function (used in various places) */
+static uint16_t ull_conn_event_counter(struct ll_conn *conn)
+{
+	struct lll_conn *lll;
+	uint16_t event_counter;
+
+	uint16_t lazy = conn->llcp.prep.lazy;
+
+	lll = &conn->lll;
+
+	/* Calculate current event counter */
+	event_counter = lll->event_counter + lll->latency_prepare + lazy;
+
+	return event_counter;
+}
+
+bool ull_conn_is_params_changed(struct ll_conn *conn, uint16_t interval,
+				uint16_t latency, uint16_t timeout)
+{
+	struct lll_conn *lll = &conn->lll;
+
+	if ((interval != lll->interval) ||
+	    (latency != lll->latency) ||
+	    (RADIO_CONN_EVENTS(timeout * 10000U,
+			       lll->interval * CONN_INT_UNIT_US) !=
+	     conn->supervision_reload)) {
+		return true;
+	}
+	return false;
+}
+
+void ull_conn_update_parameters(struct ll_conn *conn, uint8_t is_cu_proc,
+				uint8_t win_size, uint16_t win_offset_us,
+				uint16_t interval, uint16_t latency,
+				uint16_t timeout, uint16_t instant)
+{
+	struct lll_conn *lll;
+	uint32_t ticks_win_offset = 0U;
+	uint32_t ticks_slot_overhead;
+	uint16_t conn_interval_old;
+	uint16_t conn_interval_new;
+	uint32_t conn_interval_us;
+	uint8_t ticker_id_conn;
+	uint32_t ticker_status;
+	uint32_t periodic_us;
+	uint16_t latency_upd;
+	uint16_t instant_latency;
+	uint16_t event_counter;
+	uint16_t ticks_at_expire;
+
+	lll = &conn->lll;
+
+	/* Calculate current event counter */
+	event_counter = ull_conn_event_counter(conn);
+
+	instant_latency = (event_counter - instant) &
+			  0xFFFF;
+
+#if defined(CONFIG_BT_CTLR_CONN_PARAM_REQ)
```

Comment:
I would put this outside of this function - and use the API as mentioned in other comment

Reply:
or place it next to the apto/sto handling below

Reply:
I'm not sure I know what you mean?

Reply:
No, I see - looks like the comment was put on the wrong line

Reply:
I'm thinking of the handling of conn->procedure_expire, I'd move that code to where the other timeout related functionality is in this function (re. apto and sto)

Reply:
But its a nit pick!